pkg/operator/ceph/version/version.go

```diff
@@ -65,6 +66,11 @@ var (
 	// So we need to detect the build version change
 	buildVersionPattern = regexp.MustCompile(`ceph version (\d+)\.(\d+)\.(\d+)\-(\d+)`)
```

Comment:
Why not include the `commitIDPattern` here? The commit is always present.

Reply:
We can include it in the `VersionPattern` above  and not in the `buildVersionPattern`. The build number represented by `\-(\d+)` is not always present.

Reply:
Right, the overall pattern so `VersionPattern` will be better.

Reply:
checked a bit more. I don't think I can update the existing `versionPattern` or `buildVersionPattern`.  This is because the `ExtractCephVersion` function is also used to extract the ceph version from `cephCluster` CR status (via `ExtractCephVersionFromLabel()` function.  And in the `cephCluster` `status.version` does not have any commit ID. 

It looks like:
```
....
 message: Cluster created successfully
  phase: Ready
  state: Created
  storage:
    deviceClasses:
    - name: hdd
  version:
    image: ceph/ceph:v16.2.4
    version: 16.2.4-0
```


So the regex match would fail if I update `versionPattern`.